selfdrive/ui/replay/replay.cc

```diff
@@ -201,6 +194,13 @@ void Replay::stream() {
       cur_which = evt->which;
       cur_mono_time_ = evt->mono_time;
 
+      if (evt->which == cereal::Event::ROAD_ENCODE_IDX) {
+        uint64_t sof = evt->event.getRoadEncodeIdx().getTimestampSof();
+        if (sof > 0) {
+          cur_mono_time_ = sof;
```

Comment:
if we send frame based on encode_idx's mono_time, the frame will be sent later than actual time.

Reply:
and I think it’s not safe to modify cur_mono_time_ here.  it will make cur_mono_time_ old, the events sent before may be re-sent again.

Reply:
I personally recommend still using ROAD_CAMERA_STATE to send frames

Reply:
Looks like C2 has EOF set, so I'll use that as will if set.

Reply:
Would it be fine if I user evt_start_ts? Since that only seems to affect the sleeps?

Reply:
moved the timestamp handling into the Event clas